crates/goose/src/agents/agent.rs

```diff
@@ -44,9 +44,9 @@ const ESTIMATE_FACTOR_DECAY: f32 = 0.9;
 pub struct Agent {
     pub(super) provider: Arc<dyn Provider>,
     pub(super) extension_manager: Mutex<ExtensionManager>,
-    pub(super) frontend_tools: HashMap<String, FrontendTool>,
-    pub(super) frontend_instructions: Option<String>,
-    pub(super) prompt_manager: PromptManager,
+    pub(super) frontend_tools: Mutex<HashMap<String, FrontendTool>>,
+    pub(super) frontend_instructions: Mutex<Option<String>>,
```

Comment:
i would refactor this into a `frontend_tool_manager` with setter methods to set the tools and instructions (mainly to restrict the region where we acquire and release the mutex lock)